### PR TITLE
FIXED: A non-numeric value encountered on line [264|546] #3576

### PR DESCRIPTION
**Pull request by jobst**
A non-numeric value encountered in SUITECRM/include/ListView/ListViewSubPanel.php on line [264|546] #3576

There are loads of these errors when empty subpanels are encountered, empty list etc.
The proposed fixes will stop these but also reduce computing time as every stack trace/exception causes extra CPU cycles.

## Description

Replace this line (264)

   $offset = $this->getOffset($html_varName);

with this line

   $offset = empty($this->getOffset($html_varName)) ? 0:$this->getOffset($html_varName);

Replace this line (546)

    $start_record = $current_offset + 1;

with these three lines

    if( !isset($current_offset) || empty($current_offset) )                                                                    
      $current_offset=0;
    $start_record = $current_offset + 1;

## Motivation and Context

Loads of entries in log files, traces/exception wasting CPU cycles.

## How To Test This

Open any subpanel with lists or even the home page

## Types of changes

- [x ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [ x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [ ] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

**Comment by daniel-samson** [comment]
Unable to code review, as there doesn't appear to be any changes in your pull request.

Please refer to our guide on how to Contribute to SuiteCRM:
https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM

**Comment by jobst** [comment]
I just logged in as another user (second email address). I can see the code changes.

**Comment by jobst** [comment]
@daniel-samson I am still trying to sort Github out - never had so much trouble with anything! I have more changes to come as I fixed many in my SuiteCRM install, I also have a few that I have already made an "issue". Currently I have 5 PR's and it looks OK for me (I just replied to one of your comments in one of the other ones).

**Comment by horus68** [comment]
@jobst you had reverted your commits so there is no file changed to be merged
See the details below this PR title
2017-05-23 09_21_40-microsoft edge

**Comment by jobst** [comment]
I feel like an idiot in the moment ... how to I revert the revert?

**Comment by horus68** [comment]
The revert is also a commit. But I don't know if you can revert a commit after it has been pushed to a repository

Reverting a commit - https://help.github.com/desktop/guides/contributing/reverting-a-commit/

How can I fix a reverted git commit? - Stack Overflow - https://stackoverflow.com/questions/5354682/how-can-i-fix-a-reverted-git-commit

This is a must see video for Git users when they want to hit Github :)
Hitler uses git - YouTube - https://www.youtube.com/watch?v=CDeG4S-mJts

**Comment by jobst** [comment]
I was very afraid you ask me to watch that ... ;-).

**Comment by horus68** [comment]
SuiteCRM its easier then github!

**Comment by jobst** [comment]
I have reverted the revert from a reverted commit that was reverted - at least that what I thought I did.
It know shows "changed file 1"

**Comment by jobst** [comment]
Please do not ask me how I did this: "files changed 1".

File: include/ListView/ListViewSubPanel.php [hunk 1]
```diff
@@ -539,7 +539,9 @@ public function processListNavigation($smartyTemplateSection, $html_varName, $cu
             global $currentModule;
             global $app_strings;
 
-            $start_record = $current_offset + 1;
+            if( !isset($current_offset) || empty($current_offset) )
```
Comment (daniel-samson) [inline include/ListView/ListViewSubPanel.php]: PSR-2: please use curly braces ( { ... }) when defining an if statement.

See our coding standards for details http://suitecrm.com/wiki/index.php/Coding_Standards

Minor formatting issue: please remove the spaces at the start and end of the brackets:

`if(!isset($current_offset) || empty($current_offset)) `

File: include/ListView/ListViewSubPanel.php [hunk 2]
```diff
@@ -261,7 +261,7 @@ public function process_dynamic_listview_rows($data, $parent_data, $smartyTempla
             reset($data);
 
             //GETTING OFFSET
-            $offset = $this->getOffset($html_varName);
+            $offset = empty($this->getOffset($html_varName)) ? 0:$this->getOffset($html_varName);
```
Comment (daniel-samson) [inline include/ListView/ListViewSubPanel.php]: formatting issue: please change ? 0:$this->getOffset($html_varName); to 

`? 0 : $this->getOffset($html_varName);`

**Comment by Dillon-Brown** [comment]
Works as expected

File: include/ListView/ListViewSubPanel.php [hunk 3]
```diff
@@ -539,7 +539,9 @@ public function processListNavigation($smartyTemplateSection, $html_varName, $cu
             global $currentModule;
             global $app_strings;
 
-            $start_record = $current_offset + 1;
+            if(!isset($current_offset) || empty($current_offset))
```
Comment (samus-aran) [inline include/ListView/ListViewSubPanel.php]: Please stick to the format of the [Coding Standards](https://suitecrm.com/wiki/index.php/Coding_Standards#Brace_Style)

So set it as:
if(!isset($current_offset) || empty($current_offset)){
  $current_offset=0;
}

**Comment by jobst** [comment]
@samus-aran made the changes as per request.

**Pull request merged**